lib/ui/painting/image_descriptor.cc

```diff
@@ -128,4 +163,36 @@ void ImageDescriptor::instantiateCodec(Dart_Handle codec_handle,
   }
   ui_codec->AssociateWithDartWrapper(codec_handle);
 }
+
+sk_sp<SkImage> ImageDescriptor::image() const {
+  if (platform_image_generator_) {
+    SkBitmap bitmap;
+    if (!bitmap.tryAllocPixels(image_info_)) {
+      FML_LOG(ERROR) << "Failed to allocate memory for bitmap of size "
+                     << image_info_.computeMinByteSize() << "B";
+      return nullptr;
+    }
+
+    const auto& pixmap = bitmap.pixmap();
+    if (!platform_image_generator_->getPixels(
+            image_info_, pixmap.writable_addr(), pixmap.rowBytes())) {
+      FML_LOG(ERROR) << "Failed to get pixels for image.";
+      return nullptr;
+    }
+    bitmap.setImmutable();
+    return SkImage::MakeFromBitmap(bitmap);
+  }
+  return SkImage::MakeFromEncoded(buffer_);
+}
+
+bool ImageDescriptor::get_pixels(const SkPixmap& pixmap) const {
+  if (generator_) {
```

Comment:
IMO, this conditional logic can be removed in favor of having a single generator field (by using SkCodecImageGenerator as descibed below).